### DISCUSSION TITLE
Bundle `vala-lint` in `elementary-sdk`

### DISCUSSION THREAD

**Discussion by ghost** [elementary/os#670] [Ideas]
There is a problem when developing elementary apps, that is, you need to separately compile vala lint or use a docker image. That seems odd, since it's so essential for development. The other option is running a docker image but that's no good for rootless and low power systems. It's odd given it used to be an elementary os tool.

I have opened this issue in their repo https://github.com/vala-lang/vala-lint/issues/164, however I think a good step would be integrating it in Elementary first.

**Comment by danirabbit** [comment]
Looks like Vala Lint has been included in elementary-sdk for about 4 years now: https://github.com/elementary/seeds/blame/jammy/elementary-sdk#L23

**Reply by ryonakano** [reply]
It looks like [the recipe for vala-lint](https://code.launchpad.net/~elementary-os/+recipe/vala-lint-daily) exists in the daily PPA, but not in the stable PPA:

Daily:

```
user@VirtualBox-d05b9bd6:~$ cat /etc/apt/sources.list.d/elementary.list 
deb https://ppa.launchpadcontent.net/elementary-os/daily/ubuntu jammy main 
deb-src https://ppa.launchpadcontent.net/elementary-os/daily/ubuntu jammy main 

user@VirtualBox-d05b9bd6:~$ LANG=C apt search io.elementary.vala-lint
Sorting... Done
Full Text Search... Done
io.elementary.vala-lint/jammy,now 0.1+r94+pkg6~daily~ubuntu7.1 amd64 [installed]
  A Vala linter

user@VirtualBox-d05b9bd6:~$ 
```

Stable:

```
ryo@lb450m:~$ cat /etc/apt/sources.list.d/elementary.list 
deb https://ppa.launchpadcontent.net/elementary-os/stable/ubuntu jammy main 
deb-src https://ppa.launchpadcontent.net/elementary-os/stable/ubuntu jammy main 

ryo@lb450m:~$ LANG=C apt search io.elementary.vala-lint
Sorting... Done
Full Text Search... Done
ryo@lb450m:~$ 
```

which means only developers using daily image can install vala-lint from apt, but those who using stable image can't.

**Reply by danirabbit** [reply]
Ah that makes sense. As far as I know there haven't been any releases of vala-lint so I just am building a snapshot of latest master :shrug: https://code.launchpad.net/~elementary-os/+recipe/vala-lint-jammy

**Reply by ghost** [reply]
4 years 😂

**Reply by ryonakano** [reply]
@danirabbit Confirmed now I can install vala-lint on Horus, thank you!
